### PR TITLE
[ticket/15266] Fix events in content_visibility

### PR DESCRIPTION
**Pull request by javiexin**
Fixes core.phpbb_content_visibility_get_visibility_sql_before,
core.phpbb_content_visibility_get_forums_visibility_before,
core.phpbb_content_visibility_get_global_visibility_before

Does not modify the events, but fixes some comments, and changes
the rest of the function to allow (at least) the documented
event functionality.

PHPBB3-15266

Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket PHPBB3-15266:

https://tracker.phpbb.com/browse/PHPBB3-15266

### CODE REVIEW

**Comment by javiexin** [comment]
Added new function "is_visible", and replaced several immediate
uses of the above, including a single event in the new function
to handle change in all places consistently, and much simpler.

**Comment by javiexin** [comment]
Unrelated CURL timeout error.

File: phpBB/phpbb/content_visibility.php [hunk 1]
```diff
@@ -131,6 +131,42 @@ public function get_count($mode, $data, $forum_id)
...
+
+		/**
+		* Allow changing the result of calling is_visible
+		*
+		* @event core.phpbb_content_visibility_is_visible
+		* @var	bool		is_visible			Default visibility condition, to be modified by extensions if needed.
+		* @var	string		mode				Either "topic" or "post"
+		* @var	int			forum_id			Forum id of the current item
+		* @var	array		data				Array of item information
+		* @since 3.1.12-RC1
```
Comment (marc1706) [inline phpBB/phpbb/content_visibility.php]: Please change this to 3.2.2-RC1

File: phpBB/phpbb/content_visibility.php [hunk 2]
```diff
@@ -195,16 +235,21 @@ public function get_visibility_sql($mode, $forum_id, $table_alias = '')
 	*/
 	public function get_forums_visibility_sql($mode, $forum_ids = array(), $table_alias = '')
 	{
-		$where_sql = '(';
+		$where_sql = '';
 
-		$approve_forums = array_intersect($forum_ids, array_keys($this->auth->acl_getf('m_approve', true)));
+		$approve_forums = array_keys($this->auth->acl_getf('m_approve', true));
+		if ($forum_ids && $approve_forums)
```
Comment (marc1706) [inline phpBB/phpbb/content_visibility.php]: I think it's better to use `empty()` checks here.

File: phpBB/phpbb/content_visibility.php [hunk 3]
```diff
@@ -281,12 +306,12 @@ public function get_global_visibility_sql($mode, $exclude_forum_ids = array(), $
...
 		*
 		* @event core.phpbb_content_visibility_get_global_visibility_before
-		* @var	array		where_sqls							The action the user tried to execute
+		* @var	array		where_sqls							Array of extra visibility conditions. Will be joined by imploding with "OR".
 		* @var	string		mode								Either "topic" or "post" depending on the query this is being used in
 		* @var	array		exclude_forum_ids					Array of forum ids the current user doesn't have access to
 		* @var	string		table_alias							Table alias to prefix in SQL queries
 		* @var	array		approve_forums						Array of forums where the user has m_approve permissions
-		* @var	string		visibility_sql_overwrite	Forces the function to return an implosion of where_sqls (joined by "OR")
+		* @var	string		visibility_sql_overwrite	If a string, forces the function to return visibility_sql_overwrite after executing the event
```
Comment (marc1706) [inline phpBB/phpbb/content_visibility.php]: An empty string is also a string but will not force anything.

File: phpBB/phpbb/content_visibility.php (hunk 2 above)
Comment (marc1706) [reply phpBB/phpbb/content_visibility.php]: Or sizeof() / count() for that matter.

**Pull request merged**